### PR TITLE
Added isValidating and events

### PR DESCRIPTION
**Pull request by SuperCoolFrog**
Added a function named **isValidating** to **changeset**.
Added events **beforeValidation** and **afterValidation.**

### CODE REVIEW

File: addon/index.js [hunk 1]
```diff
@@ -49,7 +53,7 @@ const defaultOptions = { skipValidate: false };
 export function changeset(obj, validateFn = defaultValidatorFn, validationMap = {}, options = {}) {
   assert('Underlying object for changeset is missing', isPresent(obj));
 
-  return EmberObject.extend({
+  return EmberObject.extend(Ember.Evented, {
```
Comment (poteto) [inline addon/index.js]: Destructure please

File: addon/index.js [hunk 2]
```diff
@@ -283,6 +288,23 @@ export function changeset(obj, validateFn = defaultValidatorFn, validationMap =
...
+     * If no key is provided it will check to see if any async validator is running.
+     *
+     * @public
+     * @param  {String|Undefined} key
+     * @return {boolean}
+     */
+    isValidating(key) {
+      let runningValidations = get(this, RUNNING_VALIDATIONS);
+      let ks = emberArray(keys(runningValidations));
+      if(key) { return ks.includes(key); }
```
Comment (poteto) [inline addon/index.js]: space after `if`

File: addon/index.js [hunk 3]
```diff
@@ -483,6 +511,29 @@ export function changeset(obj, validateFn = defaultValidatorFn, validationMap =
...
+     *
+     * @private
+     * @param {String} key 
+     * @param {Boolean} value
+     */
+    _setIsValidating(key, value) {
+      let runningValidations = get(this, RUNNING_VALIDATIONS);
+      let count = get(runningValidations, key) || 0;
+
+      if(value) {
```
Comment (poteto) [inline addon/index.js]: space after `if`

File: addon/index.js [hunk 4]
```diff
@@ -483,6 +511,29 @@ export function changeset(obj, validateFn = defaultValidatorFn, validationMap =
...
+        set(runningValidations, key, count + 1);
+      }else {
```
Comment (poteto) [inline addon/index.js]: space after `}`

File: addon/index.js [hunk 5]
```diff
@@ -483,6 +511,29 @@ export function changeset(obj, validateFn = defaultValidatorFn, validationMap =
...
+        if(count === 1) {
```
Comment (poteto) [inline addon/index.js]: space after `if`

**Comment by SuperCoolFrog** [comment]
No problem and thanks for reviewing it.  I made those changes and also included in the readme documentation for the events I added.  Let me know if there is anything else.

**Pull request merged**